client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SearchDocumentationIT.java

```diff
@@ -0,0 +1,199 @@
+ * <p>
+ * You need to wrap your code between two tags like:
+ * // tag::example[]
+ * // end::example[]
+ * <p>
+ * Where example is your tag name.
+ * <p>
+ * Then in the documentation, you can extract what is between tag and end tags with
+ * ["source","java",subs="attributes,callouts,macros"]
+ * --------------------------------------------------
+ * include-tagged::{doc-tests}/SearchDocumentationIT.java[example]
+ * --------------------------------------------------
+ */
+public class SearchDocumentationIT extends ESRestHighLevelClientTestCase {
+
+    public void testScroll() throws IOException {
+        RestHighLevelClient client = highLevelClient();
+        {
+            BulkRequest request = new BulkRequest();
+            request.add(new IndexRequest("posts", "doc", "1")
+                    .source(XContentType.JSON, "title", "In which order are my Elasticsearch queries executed?"));
+            request.add(new IndexRequest("posts", "doc", "2")
+                    .source(XContentType.JSON, "title", "Current status and upcoming changes in Elasticsearch"));
+            request.add(new IndexRequest("posts", "doc", "3")
+                    .source(XContentType.JSON, "title", "The Future of Federated Search in Elasticsearch"));
+            request.setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);
+            BulkResponse bulkResponse = client.bulk(request);
+            assertSame(bulkResponse.status(), RestStatus.OK);
+            assertFalse(bulkResponse.hasFailures());
+        }
+
+        String lastScrollId = null;
+
+        {
+            // tag::search-scroll-request
+            SearchRequest searchRequest = new SearchRequest("posts"); // <1>
+            searchRequest.scroll(TimeValue.timeValueMinutes(1L)); // <2>
+            // end::search-scroll-request
+
+            searchRequest.source(new SearchSourceBuilder().size(1));
+
+            // tag::search-response-scroll-id
+            SearchResponse searchResponse = client.search(searchRequest); // <1>
+            SearchHits searchHits = searchResponse.getHits(); // <2>
+            String scrollId = searchResponse.getScrollId(); // <3>
+            // end::search-response-scroll-id
+
+            assertEquals(0, searchResponse.getFailedShards());
+            assertEquals(3L, searchResponse.getHits().getTotalHits());
+            assertEquals(1L, searchHits.getHits().length);
+            assertNotNull(scrollId);
+            lastScrollId = scrollId;
+        }
+        {
+            String scrollId = lastScrollId;
+
+            // tag::search-scroll-id
+            SearchScrollRequest scrollRequest = new SearchScrollRequest(); // <1>
+            scrollRequest.scroll(TimeValue.timeValueMinutes(1L)); // <2>
+            scrollRequest.scrollId(scrollId); // <3>
+            // end::search-scroll-id
+
+            // tag::search-scroll-execute
+            SearchResponse searchResponse = client.searchScroll(scrollRequest); // <1>
```

Comment:
shall we have the scroll execute in a loop instead? for both sync and async, so we have a complete example on how users are supposed to use it?